engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/ModCheckBase.java

```diff
@@ -98,11 +112,21 @@ protected void initialize(int startIndex, int endIndex, int checkDigitIndex, boo
 	 * @throws NumberFormatException in case character is not a digit
 	 */
 	protected int extractDigit(char value) throws NumberFormatException {
-		if ( Character.isDigit( value ) ) {
-			return Character.digit( value, DEC_RADIX );
+		if ( alphanumeric ) {
+			if ( (value >= '0' && value <= '9' ) || (value >= 'A' && value <= 'Z') ) {
+				 return value - BASE_CHAR_INDEX;
```

Comment:
This way of converting seems somewhat specific to CNPJ, I'd suggest creating an extended version of `Mod11CheckValidator` in `CNPJValidator` and overriding `extractDigit` in it.